File: Leet Code Solutions/119. Pascal's Triangle II/PralinKhaira-PascalTriangle.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        // Initialize a 2D list to represent Pascal's Triangle
        List<List<Integer>> pascalTriangle = new ArrayList<>();
        
        // Generate the rows
        for (int i = 0; i <= rowIndex; i++) {
            List<Integer> currentRow = new ArrayList<>();
            
            // The first element of each row is always '1'
            currentRow.add(1);
            
            if (i > 0) {
                // Get a reference to the previous row
                List<Integer> previousRow = pascalTriangle.get(i - 1);
                
                // Calculate and populate the middle elements of the row
                for (int j = 1; j < previousRow.size(); j++) {
                    int sum = previousRow.get(j) + previousRow.get(j - 1);
                    currentRow.add(sum);
                }
                
                // The last element of each row is also '1'
                currentRow.add(1);
            }
            
            pascalTriangle.add(currentRow);
        }
        
        return pascalTriangle.get(rowIndex);
    }
}

File: Leet Code Solutions/119. Pascal's Triangle II/PralinKhaira-PascalTriangle.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        // Initialize a 2D vector to represent Pascal's Triangle
        vector<vector<int>> pascalTriangle(rowIndex + 1);

        // Initialize the first row with a single element '1'
        pascalTriangle[0].push_back(1);

        // Generate the rest of the rows
        for (int currentRow = 1; currentRow <= rowIndex; currentRow++) {
            // The first element of each row is always '1'
            pascalTriangle[currentRow].push_back(1);

            // Get a reference to the current row and the previous row
            vector<int>& currentRowList = pascalTriangle[currentRow];
            vector<int>& previousRowList = pascalTriangle[currentRow - 1];

            // Calculate and populate the middle elements of the row
            for (int j = 1; j < previousRowList.size(); j++) {
                int sum = previousRowList[j] + previousRowList[j - 1];
                currentRowList.push_back(sum);
            }

            // The last element of each row is also '1'
            currentRowList.push_back(1);
        }

        return pascalTriangle[rowIndex];
    }
};
